### ISSUE TITLE
(Suggestion) Allow creation of legit CIAs from SD

### ISSUE THREAD

**Issue by mos6507** [d0k3/Decrypt9WIP#112]
While it is good to anonimize tickets for people wanting to spread CIAs for those only wanting personal backups it would be nice to be able to create legit CIAs where nothing has been modified and therefore all signatures are valid.

What happens if you install a CIA with anonimized ticket on a system that already contains the legit ticket? 2 different tickets for the same titleid?

**Comment by d0k3** [comment]
You know CIAs can only be legit if you actually got the content they are created from from eShop? Also, they are legit for one console only, and they may also be tied to your eShop account? Meaning: If you lose your console, or even access to the eShop, those 'legit' CIAs may be worthless.

I will consider this, but only after someone (you?) at least confirmed such legit CIAs are still installable on a fresh system with no NNID and no eShop access. You can create such CIAs using @cearp's FunkyCIA.

Also, there can be only one ticket per titleID. Meaning: most likely the genuine one would get overwritten with the illegit one. Not that it would matter much, though.

**Comment by mos6507** [comment]
> You know CIAs can only be legit if you actually got the content they are created from from eShop? Also, they are legit for one console only, and they may also be tied to your eShop account? Meaning: If you lose your console, or even access to the eShop, those 'legit' CIAs may be worthless.

This is exactly the content scenario I was thinking of this back up option for, most other installed content would have already come from a CIA (except for the few preinstalled games available) and so wouldnt need to be turned into a CIA. For safety you could always make 2 backups, 1 legit (if you enabled it) and 1 anonimized in case the worst happens. Though just losing a 3ds should be recoverable, nintendo could transfer your nnid to a new 3ds and you would be able to re-download your eShop purchases, so this would more seem useful to those getting themselves banned.

> I will consider this, but only after someone (you?) at least confirmed such legit CIAs are still installable on a fresh system with no NNID and no eShop access. You can create such CIAs using @cearp's FunkyCIA.

I have a second 3ds with no nnid but it can access eShop and FunkyCIA not an option, according to @cearp 

> '-personal', which will leave your unique ids in the CIA intact. This is so we can make perfect backups for our own system. (As opposed to wiping the ids by default, which make it installable on all consoles + requiring us to use GW or CFW to (install?) + run it) I do not know why, but when I make personal CIAs that should only work on my console, they do not install

and with limited internet I look for a method to backup already downloaded eShop content. I can decrypt the app, tmd etc and dump the ticket, so if there is a way to manually convert it to a CIA I could try that.

> Also, there can be only one ticket per titleID. Meaning: most likely the genuine one would get overwritten with the illegit one. Not that it would matter much, though.

Maybe it does not matter at the moment but nintendo could detect fake tickets in the future (maybe the already do but havent acted?), so I dont want to poison my nnid system with fake tickets. If the overwrite can be confirmed, a workaround may be to have a backup of the legit ticket and install the ticket over the fake one. How clean that would be I'm not sure. I believe there are still unknowns in the database. I'm certain I've read that deleting tickets with FBI still leaves remenents of the ticket behind. Incidentally, if only 1 ticket per titleID, why when I dump my tickets does D9 report 35 duplicate tickets?

**Comment by MelonGx** [comment]
"Legit CIA" is created by pre-installed game & SysNAND 9.3+ installable.
FunkyCIA can create that.

BTW, "Legit CIA" is not legit.
In Japanese consoles, the consoles which installed pre-installed game are recorded on Nintendo's server.
If your console has a genuine pre-installed game, once you access e-shop, it will be automatically registered on the purchase log. After that, a new ticket takes place of the old pre-installed ticket in ticket.db.
If your console doesn't have any genuine pre-installed game, even you install a "Legit CIA" on it, it will never be registered on the purchase log.

**Comment by d0k3** [comment]
Okay, can you test this? You have to select the TMD file inside the appropriate folder via the A button. The A: drive contains SysNAND titles installed to the SD, the B: drive is for EmuNAND.
https://transfer.sh/bPiPy/godmode9-20161216-024250.zip
(or build from the latest GM9 commit)

And I'll see about adding this to D9, too, but this requires testing first.

**Comment by Zetchzie** [comment]
how about cxi like dycrypt9wip be awesome to know what game im doing

**Comment by d0k3** [comment]
CXI dumping already works in GM9, if you meant that, @Zetchzie. As per this suggestion - I decided against it (for now). Most people would want anonymized CIAs (your request is more of a personal preference), so this is all that D9 will create for now. The CIA builder in D9 also requires an overhaul, too, so adding anything to the existing system is not a good idea right now.

In short: use GM9 for what you want to do :).
